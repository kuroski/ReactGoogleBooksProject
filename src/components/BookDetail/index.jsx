import React, {Component} from 'react'
import * as booksApi from '../../api'
import {FormattedMessage} from 'react-intl'

const propTypes = {
  params: React.PropTypes.object.isRequired
}

class BookDetail extends Component {
  constructor(props) {
    super(props)
    this.state = {
      book: {},
      message: ''
    }

    this.executeBookSearch = this.executeBookSearch.bind(this)
  }

  componentWillMount() {
    this.executeBookSearch(this.props.params.bookId)
  }

  executeBookSearch(bookId) {
    return booksApi.find(bookId)
      .then(response => {
      })
      .catch(error => {
        return this.setState({
          message: <FormattedMessage id="app.errors.booknotfound" defaultMessage="Nenhum livro foi encontrado"></FormattedMessage>
        })
      })
  }

  render() {
    return (
      <div>
        {this.state.message}
        Hello Book Detail {this.props.params.bookId}
      </div>
    )
  }
}

BookDetail.propTypes = propTypes

export default BookDetail