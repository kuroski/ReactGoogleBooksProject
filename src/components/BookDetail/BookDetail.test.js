import React from 'react'
import BookDetail from './'
import renderer from 'react-test-renderer'
import {shallow, mount} from 'enzyme'
import {shallowWithIntl, mountWithIntl} from '../../test/helpers/intl-enzyme-test-helper'
import bookJson from '../../test/__mocks__/book.json'
import errorBookJson from '../../test/__mocks__/errorBook.json'
import nock from 'nock'
import {API_URL, HARRY_POTTER_GET_URL, HARRY_POTTER_GET_WRONG_URL} from '../../test/__mocks__/constants'

describe('Book', () => {
  afterEach(() => {
      nock.cleanAll()
  })

  it('renders correctly', () => {
    const BookDetailTree = renderer.create(
      <BookDetail params={{bookId: bookJson.id}} />
    ).toJSON()
    expect(BookDetailTree).toMatchSnapshot()
  })

  it('should list searched book', () => {
    nock(API_URL)
      .get(HARRY_POTTER_GET_URL)
      .reply(200, bookJson)

    const wrapper = shallow(<BookDetail params={{bookId: bookJson.id}} />)
    wrapper.instance().executeBookSearch(bookJson.id)
      .then(data => {
        expect(wrapper.state('book')).toEqual(bookJson)
        expect(wrapper.state('message')).toBeNull()
      })
      .catch(error => {
        expect(error).toEqual(error)
      })
  })

  it('should throw an error when searching with an invalid ID', () => {
    nock(API_URL)
      .get(HARRY_POTTER_GET_WRONG_URL)
      .replyWithError(errorBookJson)

    const wrapper = mountWithIntl(<BookDetail params={{bookId: '123'}} />)
    // expect(wrapper.state('message').props.id).toEqual('app.errors.booknotfound')
  })
})