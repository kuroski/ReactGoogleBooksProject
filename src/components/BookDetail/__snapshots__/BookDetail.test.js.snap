exports[`Book renders correctly 1`] = `
<div>
  
  Hello Book Detail 
  K_yxDAAAQBAJ
</div>
`;
